core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -229,6 +229,9 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                 builder.indexOptions(nodeIndexOptionValue(propNode));
                 iterator.remove();
             } else if (propName.equals("include_in_all")) {
+                if(parserContext.isWithinMultiField()) {
+                    throw new MapperParsingException("sub-fields shouldn't contain property: include_in_all, and current node is:" + name);
```

Comment:
Have a look at the example below for copy_to about how we usually write these field names. We wrap all the variable bits of exception messages in `[]` almost everywhere. Right or wrong it is our habit and we should stick to it unless we have a good reason not to.